great_expectations/expectations/metrics/util.py

```diff
@@ -606,6 +606,55 @@ def column_reflection_fallback(
         return col_info_dict_list
 
 
+def get_dbms_compatible_metric_domain_kwargs(
+    metric_domain_kwargs: dict,
+    batch_columns_list: List[str | sqlalchemy.quoted_name],
+) -> dict:
+    """
+    This method checks "metric_domain_kwargs" and updates values of "Domain" keys based on actual "Batch" columns.  If
+    column name in "Batch" column list is quoted, then corresponding column name in "metric_domain_kwargs" is also quoted.
+
+    Args:
+        metric_domain_kwargs: Original "metric_domain_kwargs" dictionary of attribute key-value pairs.
+        batch_columns_list: Actual "Batch" column list (e.g., output of "table.columns" metric).
+
+    Returns:
+        metric_domain_kwargs: Updated "metric_domain_kwargs" dictionary with quoted column names, where appropriate.
+    """
+    if "column" in metric_domain_kwargs:
```

Comment:
Seems like we know the possible keys for `metric_domain_kwargs` - could we use a TypedDict? You'll find other examples if you grep the codebase

Reply:
@cdkini `metric_domain_kwargs` permeates the codebase, so I would prefer this improvement to be a separate JIRA ticket (impacting `MetricConfiguration`, `ValidationGraph`, `MetricsCalculator`, and many more core classes).  Happy to discuss.  Thanks.